config.go

```diff
@@ -32,19 +32,32 @@ func LoadLogScraperConfig(file string) (*LogScraperConfig, error) {
 
 	var cfg LogScraperConfig
 	bytes, err := ioutil.ReadFile(file)
-	if err == nil {
-		err = json.Unmarshal(bytes, &cfg)
+	if err != nil {
+		return nil, err
 	}
+
+	if err := json.Unmarshal(bytes, &cfg); err != nil {
+		return nil, err
+	}
+
+	if err := cfg.validate(); err != nil {
+		return nil, err
+	}
+
 	return &cfg, err
 }
 
-func (config *LogScraperConfig) LogSources() ([]*LogSource, []error) {
-	if config.Sources == nil || len(config.Sources) == 0 {
-		return make([]*LogSource, 0), nil
+func (cfg *LogScraperConfig) validate() error {
```

Comment:
I really don't think this function is useful. The complexity of the function is equivalent to the invocation of it. Just replace the call to validate with this body?